lib/request.js

```diff
@@ -14,37 +12,44 @@ const version = require("../package.json").version;
  * @returns {Promise}
  */
 function request(baseConfig, method, depth, xml) {
-    const config = _.assign({}, baseConfig);
-
-    config.headers = _.assign({}, baseConfig.headers);
+    const config = {
+        headers: baseConfig.headers,
+        body: xml,
+        method,
+        'headers["Content-length"]': xml.length,
+        'headers["Depth"]': depth,
+        ...baseConfig
+    };
 
     if (!baseConfig.headers || "User-Agent" in baseConfig.headers === false) {
         config.headers["User-Agent"] = "scrapegoat/" + version;
     }
 
     if (baseConfig.auth) {
-        config.auth = _.assign({}, baseConfig.auth);
-        config.auth.sendImmediately = "sendImmediately" in baseConfig.auth ? baseConfig.auth.sendImmediately : false;
+        config.auth = {
+            sendImmediately:
+                "sendImmediately" in baseConfig.auth ?
+                    baseConfig.auth.sendImmediately :
+                    false,
+            ...baseConfig.auth
+        };
     }
 
-    config.body = xml;
-    config.method = method;
-    config.headers["Content-length"] = xml.length;
-    config.headers.Depth = depth;
     config.timeout = baseConfig.timeout ? baseConfig.timeout : 10000;
 
-    return nodefn.call(doRequest, config)
-        .spread((res, body) => {
+    return new Promise((resolve, reject) => {
+        doRequest(config, (res, body) => {
```

Comment:
I was wondering if there is an error argument as well? What happens if request messes up really badly and won't even return a valid `res` at all?